src/SprykerEco/Zed/CrefoPay/Business/Oms/Command/Saver/CaptureOmsCommandSaver.php

```diff
@@ -136,8 +131,27 @@ protected function getPaymentCrefoPayTransfer(CrefoPayOmsCommandTransfer $crefoP
         $capturedAmount = $paymentCrefoPayTransfer->getCapturedAmount();
         $capturedAmount += $crefoPayOmsCommandTransfer->getRequest()->getCaptureRequest()->getAmount()->getAmount();
 
-        if ($crefoPayOmsCommandTransfer->getExpensesResponse() && $crefoPayOmsCommandTransfer->getExpensesResponse()->getIsSuccess()) {
-            $capturedAmount += $crefoPayOmsCommandTransfer->getExpensesRequest()->getCaptureRequest()->getAmount()->getAmount();
+        if ($crefoPayOmsCommandTransfer->getExpensesResponse() !== null
+            && $crefoPayOmsCommandTransfer->getExpensesResponse()->getIsSuccess()
+        ) {
+            $expensesCaptureRequest = $crefoPayOmsCommandTransfer->getExpensesRequest()->getCaptureRequest();
+            $paymentCrefoPayTransfer
+                ->setExpensesCapturedAmount(
+                    $expensesCaptureRequest
+                        ->getAmount()
+                        ->getAmount()
+                )
+                ->setExpensesCaptureId(
+                    $crefoPayOmsCommandTransfer
+                        ->getExpensesRequest()
+                        ->getCaptureRequest()
+                        ->getCaptureID()
+                );
+            $paymentCrefoPayTransfer
```

Comment:
It looks a little wrong. Why do we set data to `$paymentCrefoPayTransfer`. And on the very next row we set exact same data, but from the different source?

Reply:
Fixed